Source/WebCore/page/QuirksInternals.js

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+// @internal
+
+async function installNetflixMediaSessionQuirk() {
+    function sleepFor(duration) {
+        return new Promise(resolve => { setTimeout(resolve, duration); });
+    }
+    function getNetflixAPI() {
+        return window?.netflix?.appContext?.state?.playerApp?.getAPI();
+    }
+    function getWatchSessionId() {
+        return getNetflixAPI()?.videoPlayer?.getAllPlayerSessionIds()?.find((val => val?.includes("watch")));
+    }
+    function getVideoId() {
+        return getNetflixAPI()?.getVideoIdBySessionId(getWatchSessionId());
+    }
+    function getVideoPlayer() {
+        return getNetflixAPI()?.videoPlayer?.getVideoPlayerBySessionId(getWatchSessionId());
+    };
+
+    while (!getVideoPlayer())
+        await sleepFor(100);
```

Comment:
Should this have a timeout so it doesn't try forever they change their player code?